specification.json

```diff
@@ -10,34 +10,41 @@
         {
             "id": "Requirement 1.1.2",
             "machine_id": "requirement_1_1_2",
-            "content": "The `API` MUST provide a function to set the global `provider` singleton, which accepts an API-conformant `provider` implementation.",
+            "content": "The `API` MUST provide a function to set the default `provider`, which accepts an API-conformant `provider` implementation.",
             "RFC 2119 keyword": "MUST",
             "children": []
         },
         {
             "id": "Requirement 1.1.3",
             "machine_id": "requirement_1_1_3",
-            "content": "The `API` MUST provide a function to add `hooks` which accepts one or more API-conformant `hooks`, and appends them to the collection of any previously added hooks. When new hooks are added, previously added hooks are not removed.",
+            "content": "The `API` MUST provide a function to bind a given `provider` to one or more client `name`s.",
             "RFC 2119 keyword": "MUST",
             "children": []
         },
         {
             "id": "Requirement 1.1.4",
             "machine_id": "requirement_1_1_4",
-            "content": "The API MUST provide a function for retrieving the metadata field of the configured `provider`.",
+            "content": "The `API` MUST provide a function to add `hooks` which accepts one or more API-conformant `hooks`, and appends them to the collection of any previously added hooks. When new hooks are added, previously added hooks are not removed.",
             "RFC 2119 keyword": "MUST",
             "children": []
         },
         {
             "id": "Requirement 1.1.5",
             "machine_id": "requirement_1_1_5",
-            "content": "The `API` MUST provide a function for creating a `client` which accepts the following options:  - name (optional): A logical string identifier for the client.",
+            "content": "The API MUST provide a function for retrieving the metadata field of the configured `provider`.",
             "RFC 2119 keyword": "MUST",
             "children": []
         },
         {
             "id": "Requirement 1.1.6",
             "machine_id": "requirement_1_1_6",
+            "content": "The `API` MUST provide a function for creating a `client` which accepts the following options:  - name (optional): A logical string identifier for the client.",
+            "RFC 2119 keyword": "MUST",
+            "children": []
+        },
```

Comment:
What happens on a name collision here?
1.1.7 says that client creation MUST NOT throw or terminate abnormally. 
This results in either having to overwrite the previous client or the operation being a No-Op right?

Reply:
> This results in either having to overwrite the previous client or the operation being a No-Op right?

Ya, I think so.

I think it makes sense to overwrite it, and this should be specified. Somewhat related is [this](https://github.com/open-feature/spec/pull/183#issuecomment-1527941473) point.

Reply:
Yeah overwriting sounds good and should be documented!

Reply:
Generating a client will never overwrite, nor would it ever throw. Not sure where the confusion is on that one.

If there's a name collision on the provider registration, it would be the same behavior as a map type (e.g. overwrite). I'll update the document.